src/Interface/Modules/Render/OsprayViewerDialog.cc

```diff
@@ -341,13 +461,34 @@ void OsprayViewerDialog::createViewer(const CompositeOsprayGeometryObject& geom)
     };
 
     impl_->geoms_.clear();
-
-    for (const auto& obj : geom.objects())
-      impl_->geoms_.push_back(duplicatedCodeFromAlgorithm(obj));
-
+    
+    
+    // volumes
+    std::vector<OSPVolume> vol_list;
+    std::vector<ospcommon::range1f> voxelRange_list;
+    std::vector<ospcommon::box3f> bounds_list;
+    
+    for (const auto& obj : geom.objects()){
+      if((boost::iequals(obj->GeomType, "structVol"))||(boost::iequals(obj->GeomType, "unstructVol")) )   {
+        ospcommon::range1f r;
+        ospcommon::box3f b;
+        vol_list.push_back(duplicatedCodeFromAlgorithm_vol(obj, r, b));
+        voxelRange_list.push_back(r);
+        bounds_list.push_back(b);
+      }else
+        impl_->geoms_.push_back(duplicatedCodeFromAlgorithm(obj));
+    }
     viewer_ = new VolumeViewer(params, guiParams, { impl_->geoms_, toOsprayBox(geom.box) }, this);
-
+    
+    
     setupViewer(viewer_);
+    
+    SCIRun::LOG_DEBUG("num of vol: "+std::to_string(vol_list.size()));
```

Comment:
Not in need of change, but the logging functions are pretty smart and can deal with a lot of different types:
```suggestion
    SCIRun::LOG_DEBUG("num of vol: {}", vol_list.size());
```